### ISSUE TITLE
splines2-packages

### ISSUE THREAD

**Issue by L3vL** [ZheyuanLi/SplinesUtils#2]
Hi,

Thank you for your nice and useful package. So far your package works with 'bs()'- & 'ns()'-objects from 'splines'-package. However, the package 'splines' was replaced by the package 'splines2'. Therefore, one can't use the package with objects from this new package (especially bSpline-Object). Are you planning to update this or is there a workaround?

Thanks!

Best,
L

**Comment by ZheyuanLi** [comment]
Hello,

I don't think `splines2` has fully replaced `splines`. The package lacks an interface for predicting a fitted regression model. For example, if you fit a linear regression model containing a term constructed with `splines2`, the `predict.lm` function can not work with it.

However, I appreciate your suggestion on an extension to work with `splines2`. I will investigate this and roll out an update, hopefully within a month.

**Comment by wenjie2wang** [comment]
> However, the package 'splines' was replaced by the package 'splines2'.

@L3vL This is not true.  The package *splines* is one of the base packages, while *splines2* is intended to be a supplementary package to *splines*.

> The package lacks an interface for predicting a fitted regression model. For example, if you fit a linear regression model containing a term constructed with `splines2`, the `predict.lm` function can not work with it.

@ZheyuanLi Thanks for pointing it out.  I was not aware of it.  I will add this feature to the next version of *splines2*.

**Comment by ZheyuanLi** [comment]
@wenjie2wang At the moment your `bSpline` function does not have a dedicated class attribute. Add a class say "bSpline" to it then write a "bSpline" method for generic function `stats::makepredictcall`. You may consult `splines:::makepredictcall.bs` or `pbs:::makepredictcall.pbs` for how it looks like. Finally, don't forget to register these methods, i.e., include in your NAMESPACE file:

importFrom(stats, makepredictcall)
S3method(makepredictcall, bSpline)

**Comment by wenjie2wang** [comment]
Yes, exactly.  I figured it out by diving into the source code of `predict.lm()`, `lm()`, and all the way to `model.frame()`.

**Comment by wenjie2wang** [comment]
> At the moment your `bSpline` function does not have a dedicated class attribute. Add a class say "bSpline" to it then write a "bSpline" method for generic function `stats::makepredictcall`.

In fact, the function `bSpline()` produces an object of class `c("matrix", "bSpline2")`.  Thus, it will be straightforward to add a `makepredictcall()` method following `makepredictcall.bs()`.

**Comment by wenjie2wang** [comment]
@ZheyuanLi  Thanks again for bringing this to my attention. The package *splines2* v0.4.4 is available on CRAN, which provides the `makepredictcall()` methods as you suggested.  See the News for details: https://github.com/wenjie2wang/splines2/releases/tag/v0.4.4
